### PR TITLE
remove github IDP from vsphere build cluster

### PR DESCRIPTION
**Pull request by rvanderp3**
The intention of this PR is to remove the github IDP from the vsphere build cluster.

### CODE REVIEW

**Comment by rvanderp3** [comment]
@hongkailiu would you mind reviewing this?  I based this PR on #26216.

**Review by hongkailiu — COMMENTED** [review]
After merge, I will do 2 more clean ups:
1. The secret in Vault
2. The id/secret on GitHub

File: clusters/build-clusters/vsphere/assets/admin_cluster_oauth_template.yaml [hunk 1]
```diff
@@ -7,16 +7,6 @@ objects:
...
   spec:
     identityProviders:
-    - name: github
-      mappingMethod: claim
-      type: GitHub
-      github:
-        # client id is a secret too. https://developer.github.com/v3/guides/basics-of-authentication/
-        clientID: "${github_client_id}"
-        clientSecret:
-          name: github-client-secret
```
[the remark this answers is not in the document]
Comment (rvanderp3) [reply clusters/build-clusters/vsphere/assets/admin_cluster_oauth_template.yaml:17]: thank you @hongkailiu , great catch.  I just made the change.

**Comment by hongkailiu** [comment]
```
$ oc get secret -n ci build-farm-credentials -o yaml | yq -r '.data|keys[]'
vsphere_github_client_id
```

The secret has only one key and we do not need it any more.

```
$ oc delete secret -n ci build-farm-credentials --as system:admin
secret "build-farm-credentials" deleted
```

```
$ oc --kubeconfig ~/Documents/secret/joe2/sa.config-updater.vsphere.config delete secret -n openshift-config github-client-secret
secret "github-client-secret" deleted
```

Also deleted the item on Vault.
https://coreos.slack.com/archives/CHY2E1BL4/p1649767477091169?thread_ts=1649761829.631099&cid=CHY2E1BL4

Deleted the oauth app on GH:

Screen Shot 2022-04-12 at 9 14 31 AM

**Pull request merged**